server/apps/publish/published_item.py

```diff
@@ -174,24 +293,162 @@ def insert_into_text_archive(self, doc):
+        logging.info('Fetched transmission details for article %s' % legal_archive_doc.get('unique_name'))
+
+        output_channel_ids = list(set([str(queue_item['output_channel_id']) for queue_item in queue_items]))
+        query = {'$and': [{config.ID_FIELD: {'$in': output_channel_ids}}]}
+        output_channels = list(get_resource_service('output_channels').get(req=None, lookup=query))
+        output_channels = {str(output_channel[config.ID_FIELD]): output_channel for output_channel in output_channels}
+
+        subscriber_ids = list(set([str(queue_item['subscriber_id']) for queue_item in queue_items]))
+        query = {'$and': [{config.ID_FIELD: {'$in': subscriber_ids}}]}
+        subscribers = list(get_resource_service('subscribers').get(req=None, lookup=query))
+        subscribers = {str(subscriber[config.ID_FIELD]): subscriber for subscriber in subscribers}
+
+        for queue_item in queue_items:
+            del queue_item[config.ETAG]
+            queue_item['output_channel_id'] = output_channels[str(queue_item['output_channel_id'])]['name']
+            queue_item['subscriber_id'] = subscribers[str(queue_item['subscriber_id'])]['name']
+        logging.info(
+            'De-normalized the Transmission Detail records of article %s' % legal_archive_doc.get('unique_name'))
+
+        # De-normalizing the legal archive doc
+        self.__denormalize_user_dg_desk(legal_archive_doc)
+
+        # Get Version History
+        req = ParsedRequest()
+        req.sort = '[("%s", 1)]' % config.VERSION
+        lookup = {'$and': [{versioned_id_field(): legal_archive_doc[config.ID_FIELD]},
+                           {config.VERSION: {'$lte': legal_archive_doc[config.VERSION]}}]}
+
+        version_history = get_resource_service('archive_versions').get(req=req, lookup=lookup)
+        legal_archive_doc_versions = []
+        for versioned_doc in version_history:
+            self.__denormalize_user_dg_desk(versioned_doc)
+            del versioned_doc[config.ETAG]
+            legal_archive_doc_versions.append(versioned_doc)
+        logging.info('Fetched version history for article %s' % legal_archive_doc.get('unique_name'))
+
+        # Upserting Legal Archive
+        logging.info('Upserting Legal Archive Repo with article %s' % legal_archive_doc.get('unique_name'))
+
+        resolve_document_etag(legal_archive_doc, ARCHIVE)
+        legal_archive_component = get_component(LegalArchive)
+        lookup = {config.ID_FIELD: legal_archive_doc[config.ID_FIELD]}
+        article_in_legal_archive = legal_archive_component.find_one(lookup)
+        if article_in_legal_archive:
+            legal_archive_component.replace(lookup, legal_archive_doc)
+        else:
+            legal_archive_component.create([legal_archive_doc])
+
+        get_component(LegalArchiveVersions).create(legal_archive_doc_versions)
+        get_component(FormattedItems).create(formatted_items)
+        get_component(PublishQueue).create(queue_items)
```

Comment:
i'm a bit lost here, seems like calling create on items that exist already, can you maybe add some comment here to explain this part?